nix/build.nix

```diff
@@ -0,0 +1,153 @@
+# Building Limine with all features in Nix.
+#
+# Independent of the packing in nixpkgs, This is convenient for prototyping and
+# local development.
+#
+# These derivations always builds Limine from the current src tree and not some
+# stable release. Further, unlike the nixpkgs derivation, this derivation runs
+# the ./bootstrap step which needs network access. Due to the nature of the
+# self-hacked Git submodules download approach, packaging this project in Nix
+# is especailly complicated. The complicated multi-derivation approach below
+# is the best I can get after multiple hours of trying. :).
+
+{
+  # Helpers
+  fd
+, lib
+, keep-directory-diff
+, nix-gitignore
+, stdenv
+, stdenvNoCC
+
+  # Actual derivation dependencies.
+, autoconf
+, automake
+, cacert
+, git
+, gnumake
+, llvmPackages
+, mtools
+, nasm
+, pkg-config
+}:
+
+let
+  # The actual current source but close to the repo state, i.e., no files from
+  # the bootstrap step.
+  currentRepoSrc = nix-gitignore.gitignoreSource [
+    # Prevent that the hash ov the sources change, when we change it in Nix.
+    "flake.nix"
+    "flake.lock"
+    "nix/"
+
+    # bootstrapped sources from ./bootstrap
+    "/build-aux/freestanding-toolchain"
+    "/freestanding-headers"
+    "/decompressor/cc-runtime"
+    "/limine-efi"
+    "/tinf"
+    "/common/flanterm"
+    "/stb"
+  ] ../.;
+
+  # Contains the sources downloaded by the Git submodule-like initialation done
+  # in ./bootstrap.
+  #
+  # ALWAYS update the hash when one of the network dependencies in ./bootstrap
+  # changes.
+  # bootstrappedSrcHash = lib.fakeHash;
```

Comment:
What's this fakehash line about?

Reply:
I hope this video clarifies any issues.

TL;DR: The hash must be changed when bootstrapped sources change.

[Screencast from 2024-04-05 13-34-28.webm](https://github.com/limine-bootloader/limine/assets/5737016/cb00e68d-6044-47c5-9568-97dac5bfac02)

Reply:
Alright, makes sense.